flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ResourceManager.java

```diff
@@ -861,6 +884,8 @@ protected void closeJobManagerConnection(JobID jobId, Exception cause) {
 
 			jmResourceIdRegistrations.remove(jobManagerResourceId);
 
+			slotManager.processResourceRequirements(ResourceRequirements.empty(jobId, jobMasterGateway.getAddress()));
```

Comment:
hmm...i think the current behavior is fine for the case where a JM just crashes, but when we have HA _I think_ we are also clearing the requirements when the leader lost leadership.
Soo...`removeJob` may be a better place to put it.

Reply:
What happens in the non-HA case, where the job simply failed? Wouldn't we have to wait for 5 minutes until the resources will be freed for other jobs?

Reply:
Current PR state(`closeJobManagerConnection`):

If the _job_ fails, then resources are currently cleaned up immediately, because the JM closes the RM connection, leading to `closeJobManagerConnection`. The JM is also supposed to reduce the requirements to 0.

If the JM _crashes_, then we have to wait for the `heartbeat.timeout`.

Cleanup in removeJob:

Resources are always cleaned up after `resourcemanager.job.timeout`.


Conceptually, it seems _correct_ to wait for the job timeout; it's just that for the non-HA case this should be set to 0?

Reply:
I think I would be in favour of leaving it like it is because it gives a bit snappier behaviour wrt other jobs when a JM crashes.